src/execution/syscalls.rs

```diff
@@ -107,34 +108,33 @@ pub fn call_contract(
 //     Ok(DeployResponse { contract_address: deployed_contract_address, constructor_retdata })
 // }
 
-// TODO: EmitEvent syscall.
 // #[derive(Debug, Eq, PartialEq)]
 // pub struct EmitEventRequest {
-//     pub content: EventContent,
+//     pub keys: Vec<Felt252>,
+//     pub data: Vec<Felt252>,
 // }
 //
-// impl SyscallRequest for EmitEventRequest {
-//     // The Cairo struct contains: `keys_len`, `keys`, `data_len`, `data`·
-//     fn read(vm: &VirtualMachine, ptr: &mut Relocatable) -> SyscallResult<EmitEventRequest> {
-//         let keys =
-//             read_felt_array::<SyscallExecutionError>(vm,
-// ptr)?.into_iter().map(EventKey).collect();         let data =
-// EventData(read_felt_array::<SyscallExecutionError>(vm, ptr)?);
-//
-//         Ok(EmitEventRequest { content: EventContent { keys, data } })
-//     }
-// }
-//
-// type EmitEventResponse = EmptyResponse;
-//
-// pub fn emit_event(
-//     request: EmitEventRequest,
-//     _vm: &mut VirtualMachine,
-//     syscall_handler: &mut SyscallHintProcessor<'_>,
-//     _remaining_gas: &mut u64,
-// ) -> SyscallResult<EmitEventResponse> {
-//     Ok(EmitEventResponse {})
-// }
+
+type EmitEventRequest = EmptyResponse;
+
+impl SyscallRequest for EmitEventRequest {
+    fn read(vm: &VirtualMachine, ptr: &mut Relocatable) -> SyscallResult<EmitEventRequest> {
+        ignore_felt_array::<SyscallExecutionError>(vm, ptr)?;
+        ignore_felt_array::<SyscallExecutionError>(vm, ptr)?;
+        Ok(EmitEventRequest {})
```

Comment:
Where is this syscall defined in `cairo-lang`?

Reply:
`syscall_handler.py`:

```
    def emit_event(self, remaining_gas: int, request: CairoStructProxy) -> SyscallFullResponse:
        keys = self._get_felt_range(start_addr=request.keys_start, end_addr=request.keys_end)
        data = self._get_felt_range(start_addr=request.data_start, end_addr=request.data_end)
        self._emit_event(keys=keys, data=data)

        response_header = self.structs.ResponseHeader(gas=remaining_gas, failure_flag=0)
        return response_header, tuple()

```
and:
```
    def _emit_event(self, keys: List[int], data: List[int]):
        return
```

Reply:
That's what I thought. But then, why do you not get the `keys` and `data` arrays?

Reply:
Because emit_even in replay phase does not need them. See empty `_emit_event` above.

Reply:
TODO for what?

Reply:
We will never need to implement the functionality of the syscall directly?

Reply:
No. Here we are just replying return values in order to get contract logic behave as expected. Syscalls are reexecuted in cairo in a later stage:
![image](https://github.com/keep-starknet-strange/snos/assets/190855/cfbaae9e-ac94-4d6b-b847-c7a3e9b39da7)